framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -2367,9 +2367,15 @@ private void logBugInCF(BugInCF ce) {
             if (printClasspath) {
                 msg.add("Classpath:");
```

Comment:
As this code is building a diagnostic message, you should try to make it fail more gracefully.
How about moving this into the `then` part and adding an `else` part that mentions that Java 9+ is used?
The comment explaining the TODO should then move into the else part.

Reply:
feedback addressed